portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,102 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.gson.Gson;
+import com.google.sps.data.Marker;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Safelist;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+
+  /** Responds with a JSON array containing marker data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    Collection<Marker> markers = getMarkers();
+    Gson gson = new Gson();
+    String json = gson.toJson(markers);
+
+    System.out.println(json);
+    response.getWriter().println(json);
+  }
+
+  /** Accepts a POST request containing a new marker. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    double lat = Double.parseDouble(request.getParameter("lat"));
+    double lng = Double.parseDouble(request.getParameter("lng"));
+    // Sanitize user input to remove HTML tags and JavaScript.
+    String title = Jsoup.clean(request.getParameter("title"), Safelist.none());
+    String content = Jsoup.clean(request.getParameter("content"), Safelist.none());
+    String date = Jsoup.clean(request.getParameter("date"), Safelist.none());
+
+    Marker marker = new Marker(lat, lng, title, content, date);
+    storeMarker(marker);
+  }
+
+  /** Fetches markers from Datastore. */
+  private Collection<Marker> getMarkers() {
+    Collection<Marker> markers = new ArrayList<>();
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query = Query.newEntityQueryBuilder().setKind("Marker").build();
+    QueryResults<Entity> results = datastore.run(query);
+    while (results.hasNext()) {
+      Entity entity = results.next();
+      double lat = entity.getDouble("lat");
+      double lng = entity.getDouble("lng");
+      String title = entity.getString("title");
+      String content = entity.getString("content");
+      String date = entity.getString("date");
```

Comment:
FYI: It would be better to store this as a timestamp instead of string - https://cloud.google.com/datastore/docs/concepts/entities#date_and_time
Also don't let the user enter this field. Just use System.currentTimeMillis() to get the current timestamp on the server.